sbt_common/storage/src/test/scala/uk/ac/wellcome/storage/vhs/StringStoreVersionedHybridStoreTest.scala

```diff
@@ -106,38 +108,67 @@ class StringStoreVersionedHybridStoreTest
       }
     }
 
-    it("can store additional metadata alongside HybridRecord") {
+    describe("with metadata") {
+
       case class ExtraData(
         data: String,
         number: Int
       )
 
       val id = Random.nextString(5)
-      val record = "this goes in dynamo"
+      val record = Random.nextString(256)
 
       val data = ExtraData(
-        data = "a tragic triangle of triffids",
-        number = 6
+        data = Random.nextString(256),
+        number = Random.nextInt(256)
       )
 
-      withS3StringStoreFixtures {
-        case (_, table, hybridStore) =>
-          val future =
-            hybridStore.updateRecord(id)(record)(identity)(data)
+      it("can store additional metadata") {
+        withLocalS3Bucket { bucket =>
+          withLocalDynamoDbTable { table =>
+            withStringVHS[ExtraData, Assertion](bucket, table) { hybridStore =>
+              val future =
+                hybridStore.updateRecord(id)(record)((t, _) => t)(data)
 
-          whenReady(future) { _ =>
-            val maybeResult =
-              Scanamo.get[ExtraData](dynamoDbClient)(table.name)('id -> id)
+              whenReady(future) { _ =>
+                val maybeResult =
+                  Scanamo.get[ExtraData](dynamoDbClient)(table.name)('id -> id)
 
-            maybeResult shouldBe defined
-            maybeResult.get.isRight shouldBe true
+                maybeResult shouldBe defined
+                maybeResult.get.isRight shouldBe true
 
-            val extraData = maybeResult.get.right.get
+                val extraData = maybeResult.get.right.get
 
-            extraData.data shouldBe "a tragic triangle of triffids"
-            extraData.number shouldBe 6
+                extraData.data shouldBe data.data
+                extraData.number shouldBe data.number
+              }
+            }
           }
+        }
       }
+
+      it("provides stored metadata when updating a record") {
+        withLocalS3Bucket { bucket =>
+          withLocalDynamoDbTable { table =>
+            withStringVHS[ExtraData, Assertion](bucket, table) { hybridStore =>
+              val updatedRecord = Random.nextString(256)
+
+              val future =
+                hybridStore.updateRecord(id)(record)((t, _) => t)(data)
+
+              val updatedFuture = future.flatMap { _ =>
+                hybridStore.updateRecord(id)(updatedRecord)((t, m) =>
+                  m.toString)(data)
```

Comment:
Using `storedMetadata` for the variable name here would help to emphasise that we’re getting the metadata which is already stored.